prov/efa/src/efa_hmem.c

```diff
@@ -317,7 +335,11 @@ static int efa_domain_hmem_info_init_synapseai(struct efa_domain *efa_domain)
 	}
 
 	info->initialized = true;
-	info->p2p_disabled_by_user = false;
+	if (ofi_hmem_p2p_disabled()) {
+		info->p2p_disabled_by_user = true;
+	} else {
+		info->p2p_disabled_by_user = false;
+	}
```

Comment:
You can probably place this at the top of each function

```suggestion
	info->p2p_disabled_by_user = ofi_hmem_p2p_disabled();
```